gobblin-config-management/src/main/java/gobblin/config/configstore/impl/BaseHdfsConfigStore.java

```diff
@@ -0,0 +1,214 @@
+package gobblin.config.configstore.impl;
+
+import gobblin.config.configstore.ConfigStore;
+import gobblin.config.configstore.VersionComparator;
+import gobblin.config.utils.PathUtils;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.log4j.Logger;
+
+import com.google.common.io.Closer;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+
+/**
+ * The BaseHdfsConfigStore implements basic functions from ConfigStore which is related to HDFS
+ * @author mitu
+ *
+ */
+public abstract class BaseHdfsConfigStore implements ConfigStore {
+
+  private static final Logger LOG = Logger.getLogger(BaseHdfsConfigStore.class);
+  public static final String CONFIG_FILE_NAME = "main.conf";
+
+  private final URI storeURI;
+  private final Path location;
+  private final String currentVersion;
+  protected final FileSystem fs;
+  protected final VersionComparator<String> vc;
+
+  public BaseHdfsConfigStore(URI root) {
+    this(root, new SimpleVersionComparator());
+  }
+
+  public BaseHdfsConfigStore(URI root, VersionComparator<String> vc) {
+    try {
+      this.storeURI = root;
+      this.location = new Path(root.getPath());
+      this.fs = this.location.getFileSystem(new Configuration());
```

Comment:
I don't believe this will work as intended. If URI is hdfs://myserver:9000/path/to/config, then root.getPath = /path/to/config, so and getFileSystem will default to the one defined in Configuration regardless of the authority in the input uri.